T 2

ceasar13m opened the pull request:
Сообщения отображаются каждое со своей стороны

ceasar13m committed:
Найдены и исправлены баги при отправке сообщений.

ceasar13m committed:
Пользовательские сообщения отображаются справа, другие сообщения отображаются слева

miptildar commented on frontend/src/components/content/channels/channel/channel.jsx:
```diff
@@ -14,7 +14,7 @@ const Channel = (props) => {
              onClick={() => props.onCurrentChannel(props.channel.id)}>
             <div className={styles.img}>
-                <img src="https://i.ibb.co/YywFDHg/2020-05-04-22-39-01.jpg" alt="2020-05-04-22-39-01" border="0"/>
+                <img src="https://i.ibb.co/YywFDHg/2020-05-04-22-39-01.jpg" alt="img" border="0"/>
```
Это временное решение

miptildar commented on frontend/src/components/content/dialog/dialog.jsx:
```diff
@@ -2,31 +2,37 @@ import React, {useState} from "react";
 
-        props.sendMessage(newMessage);
+        const message = {
+            type: 'PUBLISH',
```
должна быть константа, нужно переиспользовать

ceasar13m replied:
Исправил

miptildar commented on frontend/src/components/content/dialog/messages/getDate.js:
```diff
@@ -0,0 +1,11 @@
+export const getDate = (date) => {
```
переведи эту функцию в другую папку, типа util
Назови его правильно

ceasar13m replied:
Исправил

miptildar commented on frontend/src/components/content/dialog/messages/getDate.js:
```diff
@@ -0,0 +1,11 @@
+    let year = sentDate.getFullYear();
+
+   return `${hours}:${minutes}:${seconds} ${day}.${month}.${year}`;
+}
```
это ведь чистая функция, она должна лежать в общем месте

ceasar13m replied:
Исправил

miptildar commented on frontend/src/redux/reducers/messenger_reducer.js:
```diff
@@ -1,14 +1,43 @@
+            }
+        }
+        case Types.NEW_MESSAGE: {
+            if (state.messages.get(action.payload.channelId + '') === undefined) {
```
добавление '' не выход, это считает костылем, найди другой способ
А еще лучше если все данные будут иметь стандартный формат. Приводить к формату нужно когда скачал данные.

ceasar13m replied:
Исправил

miptildar commented on src/main/java/com/ainur/broker/Worker.java:
```diff
@@ -21,13 +26,48 @@ public void run() {
+    private Pocket createPocket(Message message) {
+        Gson gson = new Gson();
+
+        Pocket pocket = new Pocket();
```
что такое Pocket?

ceasar13m replied:
Исправил

miptildar commented on src/main/java/com/ainur/broker/network/AppRESTController.java:
```diff
@@ -248,4 +258,23 @@ public ResponseEntity getAllChannels(HttpServletRequest request) {
+    @ResponseBody
+    public ResponseEntity getMessages(HttpServletRequest request) {
+        gson = new Gson();
+        log = Logger.getLogger(AppRESTController.class.getName());
```
Нет, log должен быть final static

ceasar13m replied:
Исправил

miptildar commented on src/main/java/com/ainur/broker/repository/MySQLRepository.java:
```diff
@@ -335,6 +377,19 @@ public boolean subscribe(Subscribe subscribe) throws SQLException {
 
+    public boolean isUserSubscribedToChannel(int userId, int channelId) throws SQLException {
+            Connection connection = dataSource.getConnection();
+            PreparedStatement statement = connection.prepareStatement(GET_SUBSCRIBERS);
```
Очень неэффективно, делай другой sql запрос. Есть способ проверить быстро проверить.

ceasar13m replied:
Исправил

miptildar commented on src/main/java/com/ainur/broker/repository/MySQLRepository.java:
```diff
@@ -335,6 +377,19 @@ public boolean subscribe(Subscribe subscribe) throws SQLException {
+            statement.setInt(1, channelId);
+            ResultSet resultSet = statement.executeQuery();
+            while(resultSet.next()) {
+                if(resultSet.getInt(1) == userId) {
```
Маленький метод будет скачивать кучу данных а потом проходить по ним в цикле

ceasar13m replied:
Исправил

miptildar commented on frontend/src/App.js:
```diff
@@ -6,21 +6,23 @@ import Header from "./components/content/header/header";
+import SplashScreen from "./components/loading-window/splash-screen/splash-screen";
 import SearchChannelsWindow from "./components/content/channels/searchChannelsWindow/search-channels-window";
 import styles from './App.module.css';
+import BackgroundLayer from "./components/loading-window/background-layer/background-layer";
```
test

miptildar replied:
test2

ceasar13m committed:
Исправлены ошибки после code review:
1. log сделал static final
2. все строки вынесены в константы
3. функцию, которая возвращает форматированную дату переименовал и переместил в папку util
4. Классам, которые представляют сообщения WebSocket, дал описательные названия
5. В методе isUserSubscribedToChannel() исправил SQL запрос

ceasar13m committed:
Исправил костыль с ключами Map() сообщений

ceasar13m committed:
убрал debugger'ы и console.log'и

miptildar reviewed (approved):
Выглядит хорошо, можно отправлять на мастер

ceasar13m committed:
css

ceasar13m committed:
T-2: улучшения верстки

Code chandes:

Testing:
Запуск и сборка локально
